src/text2.c

```diff
@@ -4043,8 +4043,8 @@ s32 InterpretColorChar(u8 a0)
 {
     if (a0 < 16)
         return a0;
-    if (a0 >= 48 && a0 <= 63)
-        return a0 - 48;
+    if (a0 >= '0' && a0 <= '?')
+        return a0 - '0';
```

Comment:
Wow, that is a weird comparison, but that is what it's actually doing (it parses `0123456789:;<=>?` as a color index, in order).
I wonder, does `'0' + 15` work?
My earlier suggestion meant the chars in the switch cases more, though.
Those are `H I E S T G M C D N W` in order.

Reply:
`'0' + 15` doesn't match